Prevent NPE in Txn.String()

tbg opened the pull request:
We were calling Txn.Short() on a UUID which
was potentially nil. Fixed Short() to return
a dummy string in that case.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/4371)

<!-- Reviewable:end -->

tschottdorf commented:
cc @andreimatei this is what we were seeing debugging the other day.

tamird commented:
LGTM

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[roachpb/data.go, line 803 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4371#-KATbb5MYBRKwpXAbZtL:-KATbb5NvvLZqLP-YQ3z:1018732377) ([raw file](https://github.com/cockroachdb/cockroach/blob/49d984f1ca512d6bb455eb5160cb66bc43667b39/roachpb/data.go#L803)):*
https://godoc.org/strings#Repeat so I don't have to count these by hand

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4371#-:-KATbhxEJIyKcC8EssLL:676834233)*
<!-- Sent from Reviewable.io -->

tbg committed:
Prevent NPE in Txn.String()

We were calling Txn.Short() on a UUID which
was potentially nil. Fixed Short() to return
a dummy string in that case.

tschottdorf commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[roachpb/data.go, line 803 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4371#-KATbb5MYBRKwpXAbZtL:-KATd2huKnbkDwqXT_Ik:-469593337) ([raw file](https://github.com/cockroachdb/cockroach/blob/49d984f1ca512d6bb455eb5160cb66bc43667b39/roachpb/data.go#L803)):*
done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4371)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4371)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
nice

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4371#-:-KAVlq6lPvdkyPzjqzqT:677880290)*
<!-- Sent from Reviewable.io -->

petermattis commented on roachpb/data_test.go:
```diff
@@ -415,6 +415,9 @@ func TestTransactionString(t *testing.T) {
 	}
+
+	var txnEmpty Transaction
+	_ = txnEmpty.String() // prevent regression of NPE
```
Shouldn't this be `txnEmpty.Short()`? Or is there some hidden way that this is testing the NPE that I'm not seeing?

tamird replied:
`String()` is implemented using `Short()`

petermattis replied:
Ah-ha!
